### PR TITLE
Dump nydusd panic info and backtrace into logger and show nydusd git commit version

### PR DESCRIPTION
**Pull request by changweige**
This RR helps debug and develop nydusd.
1. Nydusd is sometimes forked by nydus-snapshotter and has no stdout and stderr to print panic info. So dumping the panic info into logger is a good choice.
2. print nydusd git commit version by nydusctl CLI tool

This can address https://github.com/containerd/nydus-snapshotter/issues/104

### CODE REVIEW

**Comment by jiangliu** [comment]
The features is fantastic. Could we remove the dependency on "log-panics"?

**Review by liubogithub — APPROVED** [review]

**Comment by liubogithub** [comment]
> The features is fantastic. Could we remove the dependency on "log-panics"?

I checked with log_panics code, it's surprisingly simple, so to avoid the crate deps, maybe we can have our own `panic::set_hook(Box::new(move |info| {` in apps/lib.rs

@changweige @jiangliu what do you think?

File: src/bin/nydusd/main.rs [hunk 1]
```diff
@@ -717,6 +717,11 @@ fn main() -> Result<()> {
     let apisock = args.value_of("apisock");
 
     setup_logging(logging_file, level)?;
+
+    log_panics::Config::new()
```
Comment (imeoer) [inline src/bin/nydusd/main.rs:721]: Maybe move it into `setup_logging`, so that enables the panic hook for `nydus-image`, `nydusd`, `nydus-cached`.

Comment (changweige) [reply src/bin/nydusd/main.rs:721]: good point

**Comment by changweige** [comment]
> The features is fantastic. Could we remove the dependency on "log-panics"?

It is a little difficult. Even if we can get rid of crate `log_panics`, will still have to depend on crate `backtrace`. I prefer including the feature ASAP so we can move to other urgent tasks. If we have move bandwidth in the future, let's try to get rid of it.

**Comment by changweige** [comment]
We can get panic info in the log like
```
[2022-07-28 11:03:57.257415 +08:00] ERROR [/home/gechangwei/.cargo/registry/src/mirrors.sjtug.sjtu.edu.cn-4f7dbcce21e258a2/log-panics-2.1.0/src/lib.rs:130] thread 'fuse_server' panicked at 'assertion failed: self.offset + self.len == _offset': storage/src/device.rs:671
   0: backtrace::capture::Backtrace::create
   1: backtrace::capture::Backtrace::new

FAILED

```

**Review by imeoer — APPROVED** [review]

File: app/src/lib.rs [hunk 2]
```diff
@@ -180,6 +180,12 @@ pub fn setup_logging(log_file_path: Option<PathBuf>, level: LevelFilter) -> Resu
     }
 
     log::set_max_level(level);
+
+    // Dump panic info and backtrace to logger.
+    log_panics::Config::new()
+        .backtrace_mode(log_panics::BacktraceMode::Resolved)
+        .install_panic_hook();
```
Comment (bergwolf) [inline app/src/lib.rs:187]: Looks like this is equivalent to simply calling
```
log_panics::init()
```

ref: https://docs.rs/log-panics/latest/log_panics/

**Comment by jiangliu** [comment]
> > The features is fantastic. Could we remove the dependency on "log-panics"?
> 
> I checked with log_panics code, it's surprisingly simple, so to avoid the crate deps, maybe we can have our own `panic::set_hook(Box::new(move |info| { ... }` in apps/lib.rs
> 
> @changweige @jiangliu what do you think?

Let's move on first, and simplify it later.

File: app/src/lib.rs (hunk 2 above)
Comment (changweige) [reply app/src/lib.rs:187]: I am afraid they are not the same. I tried `log_panics::init()` ending up with no panic backtrace and resolved function name?

**Review by jiangliu — APPROVED** [review]

**Pull request merged**